src/vm/costs/mod.rs

```diff
@@ -220,57 +270,70 @@ fn parse_cost(cost_function: ClarityCostFunction, eval_result: InterpreterResult
-                    })
-                },
-                _ => Err(CostErrors::CostComputationFailed("Execution Cost tuple does not contain only UInts".to_string()))
+                    Some(UInt(read_count)),
+                ) => Ok(ExecutionCost {
+                    write_length: (*write_length as u64),
+                    write_count: (*write_count as u64),
+                    runtime: (*runtime as u64),
+                    read_length: (*read_length as u64),
+                    read_count: (*read_count as u64),
+                }),
+                _ => Err(CostErrors::CostComputationFailed(
+                    "Execution Cost tuple does not contain only UInts".to_string(),
+                )),
             }
-        },
-        Ok(Some(_)) => Err(CostErrors::CostComputationFailed("Clarity cost function returned something other than a Cost tuple".to_string())),
-        Ok(None) => Err(CostErrors::CostComputationFailed("Clarity cost function returned nothing".to_string())),
-        Err(e) => Err(CostErrors::CostComputationFailed(format!("Error evaluating result of cost function {}: {}", cost_function.get_name(), e))),
+        }
+        Ok(Some(_)) => Err(CostErrors::CostComputationFailed(
+            "Clarity cost function returned something other than a Cost tuple".to_string(),
+        )),
+        Ok(None) => Err(CostErrors::CostComputationFailed(
+            "Clarity cost function returned nothing".to_string(),
+        )),
+        Err(e) => Err(CostErrors::CostComputationFailed(format!(
+            "Error evaluating result of cost function {}: {}",
+            cost_function.get_name(),
+            e
+        ))),
     }
 }
 
 fn compute_cost(
     cost_tracker: &mut LimitedCostTracker,
     cost_function: ClarityCostFunction,
-    input_size: u64) -> Result<ExecutionCost> {
-
+    input_size: u64,
+) -> Result<ExecutionCost> {
     let mut null_store = NullBackingStore::new();
     let conn = null_store.as_clarity_db();
     let mut global_context = GlobalContext::new(conn, LimitedCostTracker::new_free());
 
-    let cost_function_reference = cost_tracker.cost_function_references
+    let cost_function_reference = cost_tracker
+        .cost_function_references
         .get(&cost_function)
-        .ok_or(CostErrors::CostComputationFailed("CostFunction not defined".to_string()))?.clone();
-
-    let cost_contract = cost_tracker.cost_contracts
-        .get(&cost_function_reference.contract_id)
-        .ok_or(CostErrors::CostComputationFailed("Cost Contract not cached".to_string()))?;
+        .ok_or(CostErrors::CostComputationFailed(
+            "CostFunction not defined".to_string(),
+        ))?
+        .clone();
 
-    let mut contract_context = ContractContext::new(cost_function_reference.contract_id.clone());
+    let cost_contract = cost_tracker
+        .cost_contracts
+        .get_mut(&cost_function_reference.contract_id)
+        .ok_or(CostErrors::CostComputationFailed(
```

Comment:
Should this be a panicking failure?  In what world will the contract be defined but its context not defined?

Reply:
The voting contract isn't implemented yet, so it's hard to say where exactly all the checks will be in the interface between this cost tracker initialization and the voting contract, but in theory, a cost function reference could be created that doesn't refer to a loaded contract. Once that check is added to the cost contract validation, yes, this should be panicking.